source/common/router/config_impl.h

```diff
@@ -1101,6 +1101,39 @@ class ConnectRouteEntryImpl : public RouteEntryImplBase {
   bool supportsPathlessHeaders() const override { return true; }
 };
 
+/**
+ * Route entry implementation for path separated prefix match routing.
+ */
+class PathSeparatedPrefixRouteEntryImpl : public RouteEntryImplBase {
+public:
+  PathSeparatedPrefixRouteEntryImpl(const VirtualHostImpl& vhost,
+                                    const envoy::config::route::v3::Route& route,
+                                    const OptionalHttpFilters& optional_http_filters,
+                                    Server::Configuration::ServerFactoryContext& factory_context,
+                                    ProtobufMessage::ValidationVisitor& validator);
+
+  // Router::PathMatchCriterion
+  const std::string& matcher() const override { return prefix_; }
+  PathMatchType matchType() const override { return PathMatchType::None; }
```

Comment:
Looking at coverage: https://storage.googleapis.com/envoy-pr/2b4b01e/coverage/source/common/router/config_impl.h.gcov.html it seems that these lines are not covered.
Can you add the similar tests as for `PrefixRouteEntryImpl`.